packages/orchestration/src/exos/icqConnectionKit.js

```diff
@@ -0,0 +1,112 @@
+// @ts-check
+/** @file ICQConnection Exo */
+import { NonNullish } from '@agoric/assert';
+import { makeTracer } from '@agoric/internal';
+import { V as E } from '@agoric/vat-data/vow.js';
+import { M } from '@endo/patterns';
+import { makeQueryPacket, parseQueryPacket } from '../utils/packet.js';
+import { ConnectionHandlerI } from '../typeGuards.js';
+
+/**
+ * @import {Zone} from '@agoric/base-zone';
+ * @import {Connection, Port} from '@agoric/network';
+ * @import {Remote} from '@agoric/vow';
+ * @import {Base64Any, RequestQueryJson} from '@agoric/cosmic-proto';
+ * @import {ResponseQuery} from '@agoric/cosmic-proto/tendermint/abci/types.js';
+ * @import {LocalIbcAddress, RemoteIbcAddress} from '@agoric/vats/tools/ibc-utils.js';
+ */
+
+const { Fail } = assert;
+const trace = makeTracer('Orchestration:ICQConnection');
+
+export const ICQMsgShape = M.splitRecord(
+  { path: M.string(), data: M.string() },
+  { height: M.string(), prove: M.boolean() },
+);
+
+export const ICQConnectionI = M.interface('ICQConnection', {
+  getLocalAddress: M.call().returns(M.string()),
+  getRemoteAddress: M.call().returns(M.string()),
+  query: M.call(M.arrayOf(ICQMsgShape)).returns(M.promise()),
+});
+
+/**
+ * @typedef {{
+ *   port: Port;
+ *   connection: Remote<Connection> | undefined;
+ *   localAddress: LocalIbcAddress | undefined;
+ *   remoteAddress: RemoteIbcAddress | undefined;
+ * }} ICQConnectionKitState
+ */
+
+/** @param {Zone} zone */
```

Comment:
I'd appreciate a more full docstring here.

In particular, let's please cite any external design constraints, such as the ICQ spec that we're conforming to.

Reply:
✅  [bd88224](https://github.com/Agoric/agoric-sdk/pull/9198/commits/bd88224e6dd025028522af2006d790601bd0d3c5)